src/Arcus.Messaging.Pumps.ServiceBus/AzureServiceBusMessagePump.cs

```diff
@@ -49,6 +60,66 @@ public abstract class AzureServiceBusMessagePump<TMessage> : MessagePump<TMessag
         /// </summary>
         protected string Namespace { get; private set; }
 
+        /// <summary>
+        /// Gets the unique identifier for this background job to distinguish this job instance in a multi-instance deployment.
+        /// </summary>
+        public string JobId { get; }
+
+        /// <summary>
+        /// Gets the name of the topic subscription; combined from the <see cref="AzureServiceBusMessagePumpSettings.SubscriptionPrefix"/> and the <see cref="JobId"/>.
+        /// </summary>
+        protected string SubscriptionName { get; }
+
+        /// <summary>
+        /// Triggered when the application host is ready to start the service.
+        /// </summary>
+        /// <param name="cancellationToken">Indicates that the start process has been aborted.</param>
+        public override async Task StartAsync(CancellationToken cancellationToken)
+        {
+            if (Settings.ServiceBusEntity == ServiceBusEntity.Topic
+                && Settings.Options.TopicSubscription.HasFlag(TopicSubscription.CreateOnStart))
+            {
+                await CreateTopicSubscriptionAsync(cancellationToken);
+            }
+
+            await base.StartAsync(cancellationToken);
+        }
+
+        private async Task CreateTopicSubscriptionAsync(CancellationToken cancellationToken)
+        {
+            ServiceBusConnectionStringBuilder serviceBusConnectionString = await GetServiceBusConnectionStringAsync();
+            var subscriptionDescription = new SubscriptionDescription(serviceBusConnectionString.EntityPath, SubscriptionName)
+            {
+                UserMetadata = $"Subscription created by Arcus job: '{JobId}' to process Service Bus messages."
+            };
+
+            var ruleDescription = new RuleDescription("Accept-All", new TrueFilter());
+            var serviceBusClient = new ManagementClient(serviceBusConnectionString);
+
+            bool subscriptionExists = 
+                await serviceBusClient.SubscriptionExistsAsync(
+                    serviceBusConnectionString.EntityPath, SubscriptionName, cancellationToken);
+            if (subscriptionExists)
```

Comment:
Let's do this before we even start composing the rule description, log it as you have it now and return. No need for an else then

Reply:
I'll move the rule description, use a try/finaly block for the service bus client to be closed, but I don't agree that removing the else and adding a return statement will benifit readability. If the return statement is forgotten, or removed by mistake, we got invalid code, while with the else this is much clearer.